graylog2-rest-client/src/main/java/org/graylog2/restclient/lib/ApiClientImpl.java

```diff
@@ -519,14 +523,22 @@ private void ensureAuthentication() {
             }
         }
 
+        private class RequestAndFuture {
```

Comment:
I'd rather not start with AutoValues here, the class is simply there to transport the request object.
Instead I'll implement it by hand quickly.